force-app/main/default/classes/HouseholdNamingService.cls

```diff
@@ -149,8 +151,17 @@ public without sharing class HouseholdNamingService {
     }
 
     private Boolean numberOfHouseholdMembersIsIncorrect(SObject household, List<Contact> householdMembers) {
-        if (householdMembers != null) {
-            return household.get('Number_of_Household_Members__c') != householdMembers.size();
+        Object currentNumberOfHouseholdMembersValue = household.get('Number_of_Household_Members__c');
```

Comment:
Is this added logic our way of guarding against not calculating the current number of houshold members?

Reply:
Yep! And with this change it now runs a recalc if members is null and Numb of HH Members is not null or zero, which recalcs the Number on the old Household when a Contact switches Households.